src/coord/src/coord.rs

```diff
@@ -5260,8 +5275,18 @@ mod timeline {
         ///
         /// This method may produce the same value multiple times, and should not be used as a test for whether
         /// a write-to-read transition has occurred, so much as an advisory signal that write capabilities can advance.
-        pub fn should_advance_to(&mut self) -> Option<mz_repr::Timestamp> {
+        pub fn should_advance_to(&mut self) -> Option<T> {
             self.advance_to.take()
         }
     }
 }
+
+pub trait CoordTimestamp: timely::progress::Timestamp {
+    fn step_forward(&self) -> Self;
+}
```

Comment:
Do you mind tossing some doccomments on the trait and the method? E.g. should `step_forward` certainly advance a thing, or panic if it cannot. I think, for example, that the implementation below that does a saturating add is not correct, as if we are at `Timestamp::MAX` doing a read, and want to switch to a write, this will provide a time that is just `Timestamp::MAX` again, not strictly advanced.

Reply:
Good point. Added docs and changed the impl to panic if it cannot step forward.